librz/parse/filter.c

```diff
@@ -310,7 +310,7 @@ static bool filter(RzParse *p, ut64 addr, RzFlag *f, RzAnalysisHint *hint, char
 					if (label) {
 						flagname = rz_str_newf(".%s", label);
 					} else {
-						flagname = strdup(f->realnames ? flag->realname : flag->name);
+						flagname = rz_str_new(flag->realname ? flag->realname : flag->name);
 					}
 					int maxflagname = p->maxflagnamelen;
 					if (maxflagname > 0 && strlen(flagname) > maxflagname) {
```

Comment:
theoretically `flagname` can be `NULL` and this check will make the process crash due a null deref because `strlen` does not check for NULL.